spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+
+
+  def change_health_check(self):
+    '''Changes the health check associated with the LB.
+    '''
+    upsert = copy.deepcopy(self.__proto_upsert)
+    hc = copy.deepcopy(self.__proto_hc)
+    hc['requestPath'] = '/hello'
+    hc['port'] = 8080
+    self._set_all_hcs(upsert, hc)
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[upsert],
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    hc_link = self._make_hc_link(hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._make_contract_clauses(contract_builder, upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='change health checks',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def change_backend_service(self):
+    '''Changes the default backend service associated with the LB.
+    '''
+    hc = copy.deepcopy(self.__proto_hc)
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    hc['name'] = 'updated-hc'
+    hc['requestPath'] = '/hello'
+    hc['port'] = 8080
+
+    bs_upsert['defaultService']['healthCheck'] = hc
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[bs_upsert],
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    hc_link = self._make_hc_link(hc['name'])
+    proto_hc_link = self._make_hc_link(self.__proto_hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._make_contract_clauses(contract_builder, bs_upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='change backend services',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def add_host_rule(self):
+    '''Adds a host rule to the url map.
+    '''
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    hr = copy.deepcopy(bs_upsert['hostRules'][0])
+    hr['hostPatterns'] = ['go.com', 'go.net']
+    hr['pathMatcher']['pathRules'][0]['paths'] = ['/go']
+    bs_upsert['hostRules'].append(hr)
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[bs_upsert],
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    hc_link = self._make_hc_link(self.__proto_hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._make_contract_clauses(contract_builder, bs_upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='add host rule',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def update_host_rule(self):
+    '''Updates a host rule to the url map.
+    '''
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    hr = copy.deepcopy(bs_upsert['hostRules'][0])
+    hr['hostPatterns'] = ['golang.com']
```

Comment:
I presume these patterns are arbitrary. Maybe name the hosts from the tests, like updated.host.com as opposed to original.host.com. This will also make it clearer when looking at logs and other state especially when a test fails and you want to know which value you have and where it came from.